server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -180,6 +182,7 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         builders.put(Names.FORCE_MERGE, new FixedExecutorBuilder(settings, Names.FORCE_MERGE, 1, -1, false));
         builders.put(Names.FETCH_SHARD_STORE,
                 new ScalingExecutorBuilder(Names.FETCH_SHARD_STORE, 1, 2 * allocatedProcessors, TimeValue.timeValueMinutes(5)));
+        builders.put(Names.SYSTEM_READ, new FixedExecutorBuilder(settings, Names.SYSTEM_READ, halfProcMaxAt5, 2000, false));
```

Comment:
Since search should be fast on these indices I wonder if a queue size of 1000 is not enough ?

Reply:
I combined the queue size for get and search here since we share the threadpool for both types of operations

Reply:
ok thanks for explaining. The query and get request are lightweight so `2000` should be ok.